test/context.type.test.ts

```diff
@@ -0,0 +1,53 @@
+import { Composer, type Context } from "../src/mod.ts";
+import {
+    assertType,
+    describe,
+    type Has,
+    type IsExact,
+    it,
+} from "./deps.test.ts";
+
+describe(
+    "context.has",
```

Comment:
We should probably call it
```suggestion
    "ctx.has* tests",
```